### PR TITLE
Improve: Index for file_map

### PR DESCRIPTION
**Pull request by windaishi**
Added two indices for file_map, that improve syncing and uploading and deleting files.

See also: https://github.com/owncloud/core/issues/6553

### CODE REVIEW

**Comment by MorrisJobke** [comment]
Ping

**Comment by karlitschek** [comment]
@DeepDiver1975 What do you think?

**Comment by DeepDiver1975** [comment]
Well - this needs some closer look - especially regarding the recent changes to the queries in the dev cycle of OC8 - @nickvergessen

File: db_structure.xml [hunk 1]
```diff
@@ -172,6 +172,24 @@
 				</field>
 			</index>
 
+			<index>
+				<name>file_map_logic_path</name>
+				<unique>false</unique>
```
Comment (nickvergessen) [inline db_structure.xml]: unique has the default false, so this is not required

File: db_structure.xml [hunk 2]
```diff
@@ -172,6 +172,24 @@
...
+				<field>
+					<name>logic_path</name>
+					<sorting>ascending</sorting>
+				</field>
+			</index>
+
+			<index>
+				<name>file_map_physic_path</name>
+				<unique>false</unique>
```
Comment (nickvergessen) [inline db_structure.xml]: same here

**Comment by nickvergessen** [comment]
@DeepDiver1975 well we still run queries against the `physic_path` and `logic_path` of that table, so I guess the indexes still make sense

**Comment by DeepDiver1975** [comment]
Let's merge this for OC8.1 - I'd like to see upgrade testing on this

**Comment by PVince81** [comment]
Is this the right time to merge this ?

**Comment by nickvergessen** [comment]
once the comments are addressed, it should be good to go

**Comment by LukasReschke** [comment]
I added @nickvergessen's changes in https://github.com/owncloud/core/pull/14171 – let's quickly review this there and get this in to ensure proper testing.

Thanks for your contribution @windaishi.

**Pull request closed without merging**